submission_worker.py

```diff
@@ -0,0 +1,309 @@
+        download_and_extract_file(annotation_file_url, phase_zip_file, phase_data_directory)
+
+    # import the challenge after everything is finished
+    challenge_module = importlib.import_module(CHALLENGE_IMPORT_STRING.format(challenge_id=challenge.id))
+    EVALUATION_SCRIPTS[challenge.id] = challenge_module
+
+
+def load_active_challenges():
+    '''
+         * Fetches active challenges and corresponding active phases for it.
+    '''
+    q_params = {}
+    q_params['start_date__lt'] = timezone.now()
+    q_params['end_date__gt'] = timezone.now()
+
+    # make sure that the challenge base directory exists
+    create_dir_as_python_package(CHALLENGE_DATA_BASE_DIR)
+
+    active_challenges = Challenge.objects.filter(**q_params)
+
+    for challenge in active_challenges:
+        phases = challenge.challengephase_set.all()
+        extract_challenge_data(challenge, phases)
+
+
+def extract_submission_data(submission_id):
+    '''
+        * Expects submission id and extracts input file for it.
+    '''
+
+    try:
+        submission = Submission.objects.get(id=submission_id)
+    except Submission.DoesNotExist:
+        print 'Submission {} does not exist'.format(submission_id)
+
+    submission_input_file = submission.input_file
+    submission_data_directory = SUBMISSION_DATA_DIR.format(submission_id=submission.id)
+    # create submission directory
+    create_dir_as_python_package(submission_data_directory)
+
+    submission_zip_file = join(submission_data_directory, 'submission_{}.zip'.format(submission.id))
+    download_and_extract_file(submission_input_file, submission_zip_file, submission_data_directory)
+
+
+def run_submission(challenge_id, phase_id, submission_id, user_annotation_file_path):
+    '''
+        * receives a challenge id, phase id and user annotation file path
+        * checks whether the corresponding evalutaion script for the challenge exists or not
+        * checks the above for annotation file
+        * calls evaluation script via subprocess passing annotation file and user_annotation_file_path as argument
+    '''
+    # evaluation_script_path = CHALLENGE_DATA_DIR.format(challenge_id=challenge_id)
+    phase_data_dir = PHASE_DATA_DIR.format(challenge_id=challenge_id, phase_id=phase_id)
+    # metadata_file_path = join(CHALLENGE_DATA_DIR.format(challenge_id=challenge_id), 'metadata')
+    submission_data_dir = SUBMISSION_DATA_DIR.format(submission_id=submission_id)
+    # create a temporary run directory under submission directory, so that
+    # main directory does not gets polluted
+    temp_run_dir = join(submission_data_dir, 'run')
+    create_dir(temp_run_dir)
+
+    # hard coded, should be fetched from challenge model
+    # execution_time_limit = 5 * 60
+
+    stdout_file_name = 'temp_stdout.txt'
+    stderr_file_name = 'temp_stderr.txt'
+
+    try:
+        submission = Submission.objects.get(id=submission_id)
+    except Submission.DoesNotExist:
+        submission = None
```

Comment:
Message should only be in queue only if its submission object has been saved. Message will be sent by an internal api, hence i think we can handle this case in the api which is responsible for creating and sending message.